### ISSUE TITLE
RejectedExecutionException from translation request

### ISSUE THREAD

**Issue by mattmikolay** [mattmikolay/pocket-ctc#2]
Received the following app crash in version 1.0:

```
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@8b1c184 rejected from java.util.concurrent.ThreadPoolExecutor@cbdcd6d[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 1]
    at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2014)
    at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:794)
    at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1340)
    at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:82)
    at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:594)
    at com.mattmik.dianma.TranslateActivity.loadInputTextFromIntent(Unknown Source)
                                            requestTranslation
                                            access$000
                                            access$100
    at com.mattmik.dianma.TranslateActivity.loadInputTextFromIntent(Unknown Source)
                                            requestTranslation
                                            access$000
                                            access$100
    at com.mattmik.dianma.TranslateActivity$2.handleMessage(Unknown Source)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:158)
    at android.app.ActivityThread.main(ActivityThread.java:7229)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

The `ExecutorService` used for translation is shut down in `onDestroy()`. Here's what I think is occurring:
1. User enters text for translation.
2. User navigates away from `TranslateActivity` and `onDestroy()` is called. (This could be due to memory pressure while `SettingsActivity` is started, or because the user is exiting the app.) Here, `mExecutor` is shut down.
3. `MSG_REQUEST_TRANSLATION` is received by `mHandler` in the dying `TranslateActivity`.
4. `requestTranslation(String inputText)` is called from within this handler and attempts to submit a new `Runnable` to `mExecutor`.

It should be possible to fix this by checking `mExecutor.isShutdown()` before submitting the new `Runnable`.

**Comment by mattmikolay** [comment]
Sure enough, this issue can be replicated easily if `UPDATE_DELAY` is increased to a value such as 2000 ms:
1. Launch app.
2. Enter text for translation.
3. Before translated result appears, press back button to destroy `TranslateActivity`.
4. Crash occurs.

```
FATAL EXCEPTION: main
Process: com.mattmik.dianma, PID: 3183
java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.FutureTask@3121f72 rejected from java.util.concurrent.ThreadPoolExecutor@8784fc3[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 0]
    at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2014)
    at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:794)
    at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1340)
    at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:82)
    at java.util.concurrent.Executors$DelegatedExecutorService.submit(Executors.java:594)
    at com.mattmik.dianma.TranslateActivity.requestTranslation(TranslateActivity.java:325)
    at com.mattmik.dianma.TranslateActivity.access$100(TranslateActivity.java:47)
    at com.mattmik.dianma.TranslateActivity$2.handleMessage(TranslateActivity.java:133)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:148)
    at android.app.ActivityThread.main(ActivityThread.java:5417)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

Modifying `requestTranslation(String inputText)` to check if the `ExecutorService` is shut down fixes this issue:

```java
private void requestTranslation(@NonNull final String inputText) {

    // Ignore translation requests if the app is shutting down
    if(mExecutor.isShutdown())
        return;

    Runnable runnable = new TranslateRunnable(mHandler, mDictionary, inputText, mTranslateMode,
            mUseTraditional);
    mExecutor.submit(runnable);

}
```

**Comment by mattmikolay** [comment]
Fixed with commit 1aa80d3. Closed.
